core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -265,6 +265,9 @@ private[deploy] class Master(
         val app = createApplication(description, driver)
         registerApplication(app)
         logInfo("Registered app " + description.name + " with ID " + app.id)
+        if(app.state == ApplicationState.WAITING) {
```

Comment:
Nit: space. The text of the warning needs to be fixed up; it's not clear what it means. Why is it abnormal to have to wait for resources? doesn't seem like a warning

Reply:
it is a warning.
Because there is no resource allocation itself may lead to spark business has been waiting, there will be problems.

Now I fix as shown below，Do you think this is correct?

 if(app.state == ApplicationState.WAITING) {
          logWarning("App need to wait worker Launch executor," +
            "maybe worker does not have extra resources to allocate")
}

Reply:
I don't think this is an abnormal case though; warn doesn't sound quite appropriate. It's already reported that it's waiting in the UI and so on.

Reply:
ok. I agree with your opinion. Thanks.